js/lib/request.js

```diff
@@ -49,8 +49,15 @@ module.exports.request = (options, callback) => {
     })
   }
 
+  let urlParsed = null
+  try {
+    urlParsed = urlParse(options.url)
```

Comment:
are you sure this fixes the crash?  `muon.url.parse` doesn't throw an error for most invalid URLs:

![screen shot 2018-04-09 at 12 50 46 pm](https://user-images.githubusercontent.com/549654/38519470-d03fca74-3bf4-11e8-86de-60201b73dc5d.png)

if the crash is because the URL is undefined, you could just replace this check with `if (typeof options.url !== 'string') { return callback(new Error('URL is not valid')) }`. then we wouldn't need to do `url.parse` on the URL unless we are in development mode (https://github.com/brave/browser-laptop/pull/13722/files#diff-2565914e73f517d565765ef02f30d306R59)